modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -164,11 +164,20 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
       }
   }
 
+  private def documentationAnnotation(hints: List[Hint]): Line = {
+    hints
+      .collectFirst { case h: Hint.Documentation => h }
+      .foldMap { doc =>
+        line"/** ${doc.docString} */"
```

Comment:
Is there another combinator that I can use to go from `Option[Hint] => Line` than `foldMap`? 
And another question, for `@documentation` traits, to I have to have a case for multiline docstrings?

Reply:
foldmap is very much adequate. 

And yes, you'll have to take consideration multimap. I've implemented something similar just yesterday for another project and it was pretty easy. Pro-tip : it's actually easier with mutations, see [here](https://github.com/build-server-protocol/build-server-protocol/pull/426/files#diff-ebe23f8ef14b3b5be88e609b4cb30ec8f88d3dd5b9817ea63cc3d0f7b4622f20R111-R116)

Reply:
Thank you! What do you mean by multimap? Is it datastructure the IR emits for things that span multiple lines?

Reply:
sorry, it's late here and I'm pretty beat, it was a typo and I meant "multiline" 😅

Reply:
So if we ignore the requirement that every line between `/**` and `*/` should start with an asterisk `*`, this also seems to print multilines fine:

```scala
  private def documentationAnnotation(hints: List[Hint]): Lines = {
    hints
      .collectFirst { case h: Hint.Documentation => h }
      .foldMap { doc =>
        Lines(line"/** ${doc.docString} */")
      }
  }
```

Result
```kotlin
/// This is the first line
/// Second line of information
/// This is a very interesting string
string AString
```

```scala
/** This is the first line
Second line of information
This is a very interesting string */
object AString extends Newtype[String] {

```

Your POC code for typescript also only renders the startin `/**` and `*/` but in smithy4s case I think it's easier to format it according to the scaladoc guidelines